src/main/java/net/sf/jabref/exporter/SaveActions.java

```diff
@@ -1,47 +1,123 @@
 package net.sf.jabref.exporter;
 
+import net.sf.jabref.Globals;
 import net.sf.jabref.MetaData;
+import net.sf.jabref.logic.cleanup.FieldFormatterCleanup;
 import net.sf.jabref.logic.formatter.BibtexFieldFormatters;
 import net.sf.jabref.logic.formatter.CaseChangers;
 import net.sf.jabref.logic.formatter.Formatter;
 import net.sf.jabref.logic.formatter.IdentityFormatter;
+import net.sf.jabref.logic.util.strings.StringUtil;
 import net.sf.jabref.model.entry.BibEntry;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 public class SaveActions {
 
-    private final Map<String, Formatter> actions;
+    private final List<FieldFormatterCleanup> actions;
 
     private List<Formatter> availableFormatters;
 
     public static final String META_KEY = "saveActions";
 
+    private boolean enabled;
+
     public SaveActions(MetaData metaData) {
-        actions = new HashMap<>();
+        Objects.requireNonNull(metaData);
+
+        actions = new ArrayList<>();
         setAvailableFormatters();
 
         List<String> formatters = metaData.getData(META_KEY);
-        if(formatters==null) {
+        if (formatters == null) {
             // no save actions defined in the meta data
             return;
         } else {
-            for (int i = 0; i < formatters.size(); i += 2) {
-                try {
-                    String field = formatters.get(i);
-                    Formatter formatter = getFormatterFromString(formatters.get(i + 1));
-
-                    actions.put(field, formatter);
-                } catch (IndexOutOfBoundsException e) {
-                    // the meta data string in the file is broken. -> Ignore the last item
-                    break;
-                }
+            parseEnabledStatus(formatters.get(0));
```

Comment:
Suggestion: change the constructor so that it accepts a boolean enabled and a list of save actions. Then the parse methods could be implemented as a static method returning a new SaveActions class.
(would make testing easier since no metadata class has to be constructed)